### ISSUE TITLE
[Issue] If base class validator function is private IsValidInput silently stops working for derived classes

### ISSUE THREAD

**Issue by johanhelsing** [dbrizov/NaughtyAttributes#187]

**Comment by IF-ACT** [comment]
@dbrizov Hi, I'm working on this issue now.

Reflection can only get declared or inherited methods, so I have previously implemented a function to get private inherited methods by recursively finding in base types, you can see my code here: [`SendMessage(object @object, string functionName)`](https://github.com/SUSTech-CANStudio/bullet-storm-unity/blob/734d446380d193b58f1ed1e6c4f588ad1871e4c6/Packages/bullet-storm-unity/Scripts/Runtime/Util/System.cs#L7)

But in NaughtyAttributes, the [`GetMethod`](https://github.com/dbrizov/NaughtyAttributes/blob/ad76a16887db5a7906cb3b132a591fb2c56fdf59/Assets/NaughtyAttributes/Scripts/Editor/Utility/ReflectionUtility.cs#L99) function calls another function named [`GetAllMethods`](https://github.com/dbrizov/NaughtyAttributes/blob/ad76a16887db5a7906cb3b132a591fb2c56fdf59/Assets/NaughtyAttributes/Scripts/Editor/Utility/ReflectionUtility.cs#L74), which returns a list of `MethodInfo`s.

- If I implement the recursive search in `GetAllMethods`, this search might be too expensive for I'm not sure where to stop searching.
- If I implement the search in `GetMethod`, the functionality of `GetMethod` and `GetAllMethods` will be different. I'm not sure if it will cause other problems or increase difficulty for understanding these methods.

Could you please give me some advice? Thanks : )

**Comment by dbrizov** [comment]
Hi. I've had a similar issue concerning private fields, but I can't remember what was the case. I resolved it by looking for the private field in the base classes as well. Check how I've implemented the `GetAllFields` or `GetAllProperties` functions and do the same with the `GetAllMethods` function. I think that will fix it, and it won't break the current behavior.
![image](https://user-images.githubusercontent.com/3619573/116098812-07cafe00-a6b4-11eb-9d1a-de3d9be94d06.png)
